### PR TITLE
[Validator] prevent double deprecation message

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Constraints/ExpressionValidatorTest.php [hunk 1]
```diff
@@ -272,7 +272,7 @@ public function testExpressionLanguageUsage()
         $this->assertTrue($used, 'Failed asserting that custom ExpressionLanguage instance is used.');
     }
 
-    /**
+    /*
```
Comment (ogizanagi) [inline src/Symfony/Component/Validator/Tests/Constraints/ExpressionValidatorTest.php]: Why this change?

File: src/Symfony/Component/Validator/Constraints/ExpressionValidator.php [hunk 2]
```diff
@@ -27,15 +27,16 @@ class ExpressionValidator extends ConstraintValidator
...
-            }
+        if (\func_num_args() > 1) {
+            @trigger_error(sprintf('The "%s" instance should be passed as "%s" first argument instead of second argument since 4.4.', ExpressionLanguage::class, __METHOD__), E_USER_DEPRECATED);
+
+            $expressionLanguage = func_get_arg(1);
 
-            if (\func_num_args() > 1 && func_get_arg(1) instanceof ExpressionLanguage) {
-                @trigger_error(sprintf('The "%s" instance should be passed as "%s" first argument instead of second argument since 4.4.', ExpressionLanguage::class, __METHOD__), E_USER_DEPRECATED);
-                $expressionLanguage = func_get_arg(1);
+            if (!$expressionLanguage instanceof ExpressionLanguage) {
```
Comment (ogizanagi) [inline src/Symfony/Component/Validator/Constraints/ExpressionValidator.php]: null is fine, so you should add `&& null !== $expressionLanguage` and update the message then.

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/ExpressionValidator.php]: 👍

File: src/Symfony/Component/Validator/Constraints/ExpressionValidator.php [hunk 3]
```diff
@@ -27,15 +27,16 @@ class ExpressionValidator extends ConstraintValidator
...
+        if (\func_num_args() > 1) {
+            @trigger_error(sprintf('The "%s" instance should be passed as "%s" first argument instead of second argument since 4.4.', ExpressionLanguage::class, __METHOD__), E_USER_DEPRECATED);
+
+            $expressionLanguage = func_get_arg(1);
 
-            if (\func_num_args() > 1 && func_get_arg(1) instanceof ExpressionLanguage) {
-                @trigger_error(sprintf('The "%s" instance should be passed as "%s" first argument instead of second argument since 4.4.', ExpressionLanguage::class, __METHOD__), E_USER_DEPRECATED);
-                $expressionLanguage = func_get_arg(1);
+            if (null !== $expressionLanguage && !$expressionLanguage instanceof ExpressionLanguage) {
+                throw new \TypeError(sprintf('Argument 2 passed to %s() must be an instance of %s, %s given.', __METHOD__, ExpressionLanguage::class, \is_object($expressionLanguage) ? \get_class($expressionLanguage) : \gettype($expressionLanguage)));
```
Comment (ogizanagi) [inline src/Symfony/Component/Validator/Constraints/ExpressionValidator.php]: Should we hint directly in the exception message to pass it as the first argument by the same way?

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/ExpressionValidator.php]: Sounds reasonable, any good idea how to phrase that?

Comment (ogizanagi) [reply src/Symfony/Component/Validator/Constraints/ExpressionValidator.php]:
```diff
- Argument 2 passed to %s() must be an instance of %s, %s given.
+ Argument 2 passed to %s() must be an instance of %s or null, %s given. Since 4.4, passing it as the second argument is deprecated and will trigger a deprecation. Pass it as first argument instead.
```

?

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/ExpressionValidator.php]: that's good 👍

**Review by ogizanagi — APPROVED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @xabbuh.
